src/components/structures/auth/Login.js

```diff
@@ -149,24 +149,24 @@ module.exports = React.createClass({
         if (!this.state.serverIsAlive) {
             this.setState({busy: true});
             // Do a quick liveliness check on the URLs
+            let aliveAgain = true;
             try {
                 await AutoDiscoveryUtils.validateServerConfigWithStaticUrls(
                     this.props.serverConfig.hsUrl,
                     this.props.serverConfig.isUrl,
                 );
                 this.setState({serverIsAlive: true, errorText: ""});
             } catch (e) {
+                const componentState = AutoDiscoveryUtils.authComponentStateForError(e);
                 this.setState({
                     busy: false,
-                    ...AutoDiscoveryUtils.authComponentStateForError(e),
+                    ...componentState,
                 });
-                if (this.state.serverErrorIsFatal) {
-                    return; // Server is dead - do not continue.
-                }
+                aliveAgain = !componentState.serverErrorIsFatal;
             }
 
             // Prevent people from submitting their password when something isn't right.
-            if (!this.state.serverIsAlive) {
+            if (!aliveAgain) {
```

Comment:
this seems to change what property is checked? from `serverIsAlive` to `serverErrorIsFatal`. Are they equivalent in this case?

Reply:
They aren't equivalent, and it was actually a bug in the old diff. `serverIsAlive` may be false when the identity server is non-responsive, but that shouldn't prevent us from logging in. `serverErrorIsFatal` is set when there's no possible way for the auth to continue (homeserver is dead).